lua/SS13.lua

```diff
@@ -65,12 +65,25 @@ function SS13.register_signal(datum, signal, func, make_easy_clear_function)
 	local path = {"SS13", "signal_handlers", ref, signal, callback_ref, "func"}
 	callback:set_var("arguments", {path})
 	if not SS13.signal_handlers[ref]["_cleanup"] then
-		local cleanup_path = {"SS13", "signal_handlers", ref, "_cleanup"}
+		local cleanup_path = {"SS13", "signal_handlers", ref, "_cleanup", "func"}
 		local cleanup_callback = SS13.new("/datum/callback", SS13.state, "call_function_return_first", cleanup_path)
 		cleanup_callback:call_proc("RegisterSignal", datum, "parent_qdeleting", "Invoke")
-		SS13.signal_handlers[ref]["_cleanup"] = function(datum)
+		SS13.signal_handlers[ref]["_cleanup"] = {func=function(datum)
```

Comment:
Split this out so it looks like:

```lua
whatever = {
    func = function(datum)
        -- stuff
    end,

    callback = cleanup_callback
}
```

(Bonus points if you just install the StyLua extension and run it over all of this)

...also, what does callback actually do here that can't be done afterwards?

Reply:
The callback field is kept as a reference so that we can find its index in the datum's `comp_lookup[COMSIG_PARENT_QDELETING]` list, in order to move it to the very end of the list.

Reply:
i see